airflow/executors/kubernetes_executor.py

```diff
@@ -318,7 +320,8 @@ def __init__(self,
         self.kube_watcher = self._make_kube_watcher()
 
     def _make_kube_watcher(self) -> KubernetesJobWatcher:
-        resource_version = KubeResourceVersion.get_current_resource_version()
+        global global_resource_version  # pylint: disable = global-statement
```

Comment:
Does this actually need to be global? Even a class-level variable would be better than an actual global, but an instance property would be best still if that's possible -- like couldn't this be stored as a property of the KubernetesExecutor, say?

Reply:
@ashb the hard art is the fact that the K8sExecutor is split across 3 separate classes. I can't figure out if a Singleton would be more or less antipattern.

Reply:
@ashb replaced with a Singleton